### ISSUE TITLE
Elm vs TypeScript?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/learn-elm#20]
asked by @stevepeak in chat ...
![elm-vs-typescript](https://cloud.githubusercontent.com/assets/194400/20314694/f3e944b6-ab52-11e6-87b2-7834477b1444.png)

> relevant to @Kumjami, @jruts & @Talento90 who are all `.ts` _aficionados_. 😉

**Comment by Talento90** [comment]
Unfortunately have never tried Elm so I can't compare them :(

I found this topic that can be useful for this discussion: [](https://gist.github.com/NickClark/a13139a49e083af88405)

**Comment by nelsonic** [comment]
@Talento90 thanks _very_ much for sharing that link! ❤️ 
_hopefully_ once our `learn-elm` is _`#ready`_ we will be in a position to make a _fair_ comparison. 😉

**Comment by gabrielperales** [comment]
From my point of view, TS is just a super set of ES6, we could call it ESX or whatever. It is cool because it should be easy to learn if you already know ES6, and you can be sure your code will be more robust than with just ES6.

The cool thing about Elm is that is a pure functional language. It is a good way to learn more about that paradigm, and also your code will be more robust (more than if you were using TS!) because it is also typed, but it also handle *undefined* and *null* with the *Maybe* type, and that is awesome.

**Comment by nelsonic** [comment]
@gabrielperales agree, TS is a _superset_ that allows JS people to add _useful_ type annotations. 👍 
We _used_ TS on a couple of client projects and the team at the client _loved_ it.
The reason I _prefer_ using Elm is because it's a **_much_ smaller language** 
This page of how to define variables is one of the _many_ reasons ***why*** I prefer Elm:
https://www.typescriptlang.org/docs/handbook/variable-declarations.html

I 😍  the fact that Elm has `let` but no `var` or `const` and that all variables are _immutable_ by `default` without needing to _import_ ***another*** library like `immutable.js` or `mori`.

Elm's _advantages_ are not limited to Type annotations, immutability and reduced (_but still complete_) language feature.

Perhaps the _best_ description of "***Why Elm***" is the "**6 Months With Elm** talk: https://youtu.be/R2FtMbb-nLs?t=47m36s

**Comment by Talento90** [comment]
As I said before I have never tried elm but how about ScalaJs? It is functional and typed language! It seems to be a third option!

**Comment by newswim** [comment]
Elm vs. [PureScript](http://www.purescript.org) is probably a bit more 🍎 's to 🍎 's

_Both_ are heavily influenced by the [Haskell language](https://www.haskell.org/), from where we get a lot of the principles we often cite when talking about Elm..

- Lazy evaluation
- Pure functions
- Fast, immutable data structures
- Type declaration

On that note, there are some [complaints](http://reasonablypolymorphic.com/blog/elm-is-wrong) that Elm doesn't support custom types.

Purescript is definitely gaining some traction. There are a few videos on Egghead.io and I know that Christopher Allen (@bitemyapp), author of [Haskell Book](http://haskellbook.com/) seems to be a fan, ([judging by his repos](https://github.com/bitemyapp?language=purescript&tab=stars)).

**Comment by bitemyapp** [comment]
@newswim I use GHCJS at work, not PureScript.

Please don't invoke my name in favor of languages just because I forked some repositories.

**Comment by newswim** [comment]
@bitemyapp My apologies, I didn't mean to speak for you, my hope was just to document this space more thoroughly. I thought I saw a video where you were speaking of it positively, but then couldn't find it. My mistake, again.
